### PR TITLE
[TT-XLA][PyTorch] Training Workload for Llama 3.2 1B - Adapter Layers

### PR DESCRIPTION
**Pull request by mmilosevicTT**
### Ticket
Closes https://github.com/tenstorrent/tt-blacksmith/issues/234

### Model Training Setting

- **Model:** `Llama 3.2 1B`
- **Strategy:** SFT with Adapter Layers on a single chip (N150)
- **Framework:** PyTorch
- **Task:** Sentiment prediction using `SST2`

### What's Changed

This PR touches on the following things:
- Refactored `llama` experiment to support different training methods (apart from LoRA).
- Implemented `llama` with Adapter Layers training workload through TT-XLA.
- Added functionality for loading model from huggingface with adapter layers.
- Moved common requirements to `requirements.txt`

### Experiment Results

For completeness here is a train/loss curve of the experiment runs during testing.

W B Chart 02_12_2025, 12_31_16

### Checklist
- [x] Main training workflow implementation
- [x] CPU testing
- [x] GPU testing
- [x] TT device testing

### CODE REVIEW

File: blacksmith/experiments/torch/llama/xla/test_llama_fine_tuning_pure_torch.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def train(config: TrainingConfig, device: torch.device, logger: TrainingLogger,
     if args.config:
         config_file_path = args.config
     else:
-        config_file_path = os.path.join(os.path.dirname(__file__), "test_llama_fine_tuning_pure_torch.yaml")
+        config_file_path = "blacksmith/experiments/torch/llama/xla/lora/test_lora.yaml"
     config = generate_config(TrainingConfig, config_file_path)
```
[the remark this answers is not in the document]
Comment (pmarkovicTT) [reply blacksmith/experiments/torch/llama/xla/test_llama_fine_tuning_pure_torch.py]: My suggestion is to use parser for args and fix configs once my PR for deviceManager lands. I think we can start moving towards using one script and multiple configs
